mr/src/main/java/org/elasticsearch/hadoop/util/IOUtils.java

```diff
@@ -209,4 +213,33 @@ private static BytesArray unwrapStreamBuffer(InputStream in) {
         }
         return null;
     }
+
+    /**
+     * Convert either a file or jar url into a local canonical file, or null if the file is a different scheme.
+     */
+    public static String toCanonicalFilePath(URL fileURL) throws URISyntaxException, IOException {
+        if (fileURL == null) {
+            return null;
+        }
+
+        // Only handle jar: and file: schemes
+        if (!fileURL.getProtocol().equals("jar") && !fileURL.getProtocol().equals("file")) {
```

Comment:
nit: prefer "constant".equals over .equals("constant") to avoid NPE .. which is not an issue here just a good practice.